src/game/GameLogicModel.js

```diff
@@ -0,0 +1,112 @@
+import CONSTANT from "../constant.js";
+import CustomError from "../CustomError.js";
+import validation from "../validation.js";
+
+export default class GameLogicModel {
+  constructor() {
+    this.count = 0;
+    this.randomNumber = 0;
+    this.prevValue = [];
+    this.isReStart = true;
+    this.inputValue = null;
+    this.game = null;
+  }
+
+  #getRandomNumber() {
+    return Math.floor(Math.random() * CONSTANT.NUMBER.MAX_NUMBER) + 1;
+  }
+
+  #updateCount(count) {
+    this.count = count;
+  }
+
+  #updatePrevValue(value) {
+    this.prevValue = value;
+  }
+
+  #updateIsReStart(value) {
+    this.isReStart = value;
+  }
+
+  #updateInputValue(value) {
+    this.inputValue = value;
+  }
+
+  validateNumberInput(number) {
+    if (validation.main(CONSTANT.ID.NUMBER, number, this.prevValue)) {
+      this.sameErrorMessage(number);
+    }
+    this.#updateInputValue(number);
+  }
+
+  reTryGame(retryValue) {
+    const isRetry = validation.main(CONSTANT.ID.RETRY, retryValue);
+    this.#updateIsReStart(isRetry);
+    !isRetry && this.reTryErrorMessage(retryValue);
+  }
+
+  reTryErrorMessage(inputValue) {
+    if (!validation.validateInput(inputValue, CONSTANT.MESSAGE.NO)) {
+      throw new CustomError(CONSTANT.ID.RETRY, "입력 형식이 잘못되었습니다.");
+    }
+  }
+
+  sameErrorMessage(inputValue) {
+    if (validation.sameValidation(inputValue, this.prevValue)) {
+      throw new CustomError(
+        CONSTANT.ID.NUMBER,
+        "동일한 값을 입력하였습니다.다시 입력해주세요"
+      );
+    } else {
+      throw new CustomError(
+        CONSTANT.ID.NUMBER,
+        "입력 형식이 잘못되었습니다.다시 입력해주세요"
+      );
+    }
```

Comment:
P5;

아래와 같이 간결하게 바꿔볼 수 있을 것 같아요.

```suggestion
	const errorMessage = validation.sameValidation(inputValue, this.prevValue) ? "동일한 값을 입력하였습니다.다시 입력해주세요" : "입력 형식이 잘못되었습니다.다시 입력해주세요"
	
	throw new CustomError(
		CONSTANT.ID.NUMBER,
		errorMessage
	);
```